File: controllers/dashboard_controller.py
from flask import Blueprint, render_template, request, redirect, url_for, flash
from flask_login import login_required, current_user
from services import FTPService, TransferService

dashboard_bp = Blueprint('dashboard', __name__)

@dashboard_bp.route('/dashboard')
@login_required
def index():
    ftp_accounts = FTPService.get_user_accounts(current_user.id)
    recent_logs = TransferService.get_user_transfers(current_user.id, 10)
    return render_template('dashboard.html', ftp_accounts=ftp_accounts, recent_logs=recent_logs)

# FTP hesabı ekleme özelliği kaldırıldı - sadece admin ekleyebilir

@dashboard_bp.route('/browse/<int:account_id>')
@login_required
def browse_files(account_id):
    account = FTPService.get_account_by_id(account_id)
    
    if not account:
        flash('FTP hesabı bulunamadı', 'error')
        return redirect(url_for('dashboard.index'))
    
    # Check if user has access to this account
    assigned_users = FTPService.get_account_assignments(account_id)
    if current_user not in assigned_users:
        flash('Bu hesaba erişim yetkiniz yok', 'error')
        return redirect(url_for('dashboard.index'))
    
    try:
        files = []
        if account.protocol == 'ftp':
            files, error = FTPService.list_ftp_files(account)
        elif account.protocol == 'sftp':
            files, error = FTPService.list_sftp_files(account)
        else:
            flash('Desteklenmeyen protokol', 'error')
            return redirect(url_for('dashboard.index'))
        
        if error:
            flash(f'Dosya listesi alınırken hata oluştu: {error}', 'error')
            return redirect(url_for('dashboard.index'))
        
        return render_template('browse.html', account=account, files=files)
    except Exception as e:
        flash(f'Dosya listesi alınırken hata oluştu: {str(e)}', 'error')
        return redirect(url_for('dashboard.index'))

File: src/main/java/com/xfer/controller/DashboardController.java
package com.xfer.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.Authentication;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

import com.xfer.entity.FTPUserAssignment;
import com.xfer.entity.User;
import com.xfer.repository.FTPUserAssignmentRepository;
import com.xfer.service.FTPService;
import com.xfer.service.TransferService;

@Controller
@RequestMapping("/dashboard")
public class DashboardController {
    
    @Autowired
    private FTPService ftpService;
    
    @Autowired
    private FTPUserAssignmentRepository ftpUserAssignmentRepository;
    
    @Autowired
    private TransferService transferService;
    
    @GetMapping
    public String dashboard(Model model, Authentication authentication, 
                          @RequestParam(value = "error", required = false) String error) {
        User currentUser = (User) authentication.getPrincipal();
        
        // Get user's assigned FTP accounts
        List<com.xfer.entity.FTPAccount> ftpAccounts = ftpService.getUserAccounts(currentUser.getId());
        
        // Get recent transfer logs
        List<com.xfer.entity.TransferLog> recentLogs = transferService.getRecentUserTransferLogs(currentUser.getId(), 10);
        
        model.addAttribute("ftpAccounts", ftpAccounts);
        model.addAttribute("recentLogs", recentLogs);
        model.addAttribute("currentUser", currentUser);
        
        // Handle error messages
        if ("access_denied".equals(error)) {
            model.addAttribute("error", "Bu sayfaya erişim yetkiniz yok. Sadece admin kullanıcıları erişebilir.");
        }
        
        return "dashboard";
    }
    
    @GetMapping("/browse/{accountId}")
    public String browseFiles(@PathVariable Long accountId, 
                             @RequestParam(required = false) String path, 
                             Model model, Authentication authentication) {
        User currentUser = (User) authentication.getPrincipal();
        
        // Check if user has access to this account
        if (!ftpService.hasUserAccess(accountId, currentUser.getId())) {
            model.addAttribute("error", "Bu hesaba erişim yetkiniz yok");
            return "redirect:/dashboard";
        }
        
        try {
            // Get account details
            com.xfer.entity.FTPAccount account = ftpService.getAccountById(accountId)
                    .orElseThrow(() -> new RuntimeException("FTP hesabı bulunamadı"));
            
            System.out.println("Browse request - Account ID: " + accountId + ", Path: " + path);
            
            // Get files list
            List<FTPService.FileInfo> files = ftpService.listFiles(accountId, path);
            
            // Get user's assignment for this account
            FTPUserAssignment userAssignment = null;
            if (!"admin".equals(currentUser.getRole())) {
                List<FTPUserAssignment> assignments = ftpUserAssignmentRepository.findByFtpAccountIdWithUser(accountId);
                userAssignment = assignments.stream()
                        .filter(assignment -> assignment.getUserId().equals(currentUser.getId()))
                        .findFirst()
                        .orElse(null);
            }
            
        model.addAttribute("account", account);
        model.addAttribute("files", files);
        model.addAttribute("currentUser", currentUser);
        model.addAttribute("assignment", userAssignment);
        model.addAttribute("currentPath", path != null ? path : "/");
            
            return "browse";
            
        } catch (Exception e) {
            model.addAttribute("error", "Dosya listesi alınırken hata: " + e.getMessage());
            return "redirect:/dashboard";
        }
    }
}
